job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -160,19 +164,33 @@ private static void migrate(MigrateCommand command, WritePType writeType, FileSy
         CreateFilePOptions.newBuilder().setWriteType(writeType).build();
     OpenFilePOptions openFileOptions =
         OpenFilePOptions.newBuilder().setReadType(ReadPType.NO_CACHE).build();
-    try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
-         FileOutStream out = fileSystem.createFile(new AlluxioURI(destination), createOptions)) {
+    final AlluxioURI destinationURI = new AlluxioURI(destination);
+    boolean retry;
+    do {
+      retry = false;
       try {
-        IOUtils.copyLarge(in, out, new byte[8 * Constants.MB]);
-      } catch (Throwable t) {
-        try {
-          out.cancel();
-        } catch (Throwable t2) {
-          t.addSuppressed(t2);
+        try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
+             FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
```

Comment:
Can we merge the two try-blocks?

Reply:
You can't. The first try block can't be merged with the 2nd try block because the 2nd try block doesn't reach the catch block when the try statement fails. The 2nd try block can't be merged with the 3rd try block because out variable will not be in scope in the catch statement.